app/src/org/commcare/dalvik/activities/CommCareVerificationActivity.java

```diff
@@ -56,19 +56,51 @@
      */
     private boolean newMediaToValidate = false;
 
+    /**
+     * Indicates whether this activity was launched from the AppManagerActivity
+     */
+    private boolean fromManager;
+
     public void onCreate(Bundle savedInstanceState){
+
         super.onCreate(savedInstanceState);
         
         setContentView(R.layout.missing_multimedia_layout);
         
         retryButton = (Button)findViewById(R.id.screen_multimedia_retry);
         retryButton.setOnClickListener(this);
+                
+        this.fromManager = this.getIntent().
+        		getBooleanExtra(AppManagerActivity.KEY_LAUNCH_FROM_MANAGER, false);
+        if (fromManager) {
+            Button skipButton = (Button)findViewById(R.id.skip_verification_button);
+            skipButton.setVisibility(View.VISIBLE);
+            skipButton.setOnClickListener(this);
+        }
         
         missingMediaPrompt = (TextView)findViewById(R.id.MissingMediaPrompt);
         
         fire();
     }
     
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        // Workaround for the possibility that CommCare screen is left off in the
+        // VerificationActivity, but then something is done on the Manager screen that means we
+        // no longer want to be there
+        boolean shouldBeHere = (CommCareApplication._().getVisibleAppRecords().size() == 1 &&
+                CommCareApplication._().getReadyAppRecords().size() == 0);
+
+        // If not, and we were not explicitly sent here from the Manager screen, then redirect to
+        // CommCareHomeActivity
+        if (!fromManager && !shouldBeHere) {

```

Comment:
If we got here from the home activity, we're actually _returning_ to it, aren't we? We'd want to call finish() instead of startActivity